### ISSUE TITLE
Data Explorer: rename and refactor

### ISSUE THREAD

**Issue by stormpython** [nteract/archived-desktop-app#4019]
**Is your feature request related to a problem? Please describe.**
Create and refactor the `Data Explorer` package.

**Describe the solution you'd like**
The `data-transform` should be renamed to `data-explorer`. There are some performance issues with the Typescript and Styled Components transition that need to be fixed. As well as some general code maintenance and cleanup.

[WIP] The following issues need to be created:
- [x] **(Merged)** [PR #4065] Rename `transform-dataresource` to `data-explorer`.
- [x] **(Merged)** [PR #4091] Fix styled components/CSS perf issues.
- [x] [PR #4113] General CSS cleanup

**Comment by stormpython** [comment]
@BenRussert @emeeks feel free to add to the list above. This can be the main issue around the data explorer refactor and we can create sub issues from here. I have more items listed, but these were the top 3 issues to get this part done quickly and we can do more general refactoring later.

**Comment by BenRussert** [comment]
The perf issue is definitely coming from `@nteract/styled-blueprintjsx`, but I have no idea how to fix it without switching to a css loader.

I put together a create-react-app prototyping package that renders the dx component with the iris data set. It is hosted here if you want to check it out: https://data-explorer-web.firebaseapp.com/. If you check devtools, I passed a logger to `onMetadataChange`  so we can see the changes in the console.

The project relies on a monorepo branch where I removed `blueprint-styledjsx` to make the app responsive. [Here is a link to the project source ](https://github.com/benrussert/data-explorer-web).
